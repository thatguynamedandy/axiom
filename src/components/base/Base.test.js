import React, { Component } from 'react';
import renderer from 'react-test-renderer';
import { Base } from 'bw-axiom';

function getComponent(props = {}) {
  return renderer.create(
    <Base { ...props } />
  );
}

describe('Base', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders string Component', () => {
    const component = getComponent({ Component: 'span' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders class Component', () => {
    class CustomComponent extends Component {
      render() {
        return (
          <span />
        );
      }
    }

    const component = getComponent({ Component: CustomComponent });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders functional Component', () => {
    function CustomComponent() {
      return (
        <span />
      );
    }

    const component = getComponent({ Component: CustomComponent });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders className', () => {
    const component = getComponent({ className: 'Lorem Ipsum' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders hiddenUntil always', () => {
    const component = getComponent({ hiddenUntil: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders hiddenUntil passed small breakpoint', () => {
    const component = getComponent({ hiddenUntil: 'small' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders hiddenUntil passed medium breakpoint', () => {
    const component = getComponent({ hiddenUntil: 'medium' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders hiddenUntil passed large breakpoint', () => {
    const component = getComponent({ hiddenUntil: 'large' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders visibleUntil always', () => {
    const component = getComponent({ visibleUntil: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders visibleUntil passed small breakpoint', () => {
    const component = getComponent({ visibleUntil: 'small' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders visibleUntil passed medium breakpoint', () => {
    const component = getComponent({ visibleUntil: 'medium' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders visibleUntil passed large breakpoint', () => {
    const component = getComponent({ visibleUntil: 'large' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textBreak none', () => {
    const component = getComponent({ textBreak: 'none' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textBreak all', () => {
    const component = getComponent({ textBreak: 'all' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textBreak word', () => {
    const component = getComponent({ textBreak: 'word' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textCase upper', () => {
    const component = getComponent({ textCase: 'upper' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textCase capital', () => {
    const component = getComponent({ textCase: 'capital' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textCase lower', () => {
    const component = getComponent({ textCase: 'lower' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textCenter always', () => {
    const component = getComponent({ textCenter: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textCenter passed small breakpoint', () => {
    const component = getComponent({ textCenter: 'small' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textCenter passed medium breakpoint', () => {
    const component = getComponent({ textCenter: 'medium' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textCenter passed large breakpoint', () => {
    const component = getComponent({ textCenter: 'large' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textLeft always', () => {
    const component = getComponent({ textLeft: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textLeft passed small breakpoint', () => {
    const component = getComponent({ textLeft: 'small' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textLeft passed medium breakpoint', () => {
    const component = getComponent({ textLeft: 'medium' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textLeft passed large breakpoint', () => {
    const component = getComponent({ textLeft: 'large' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textRight always', () => {
    const component = getComponent({ textRight: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textRight passed small breakpoint', () => {
    const component = getComponent({ textRight: 'small' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textRight passed medium breakpoint', () => {
    const component = getComponent({ textRight: 'medium' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textRight passed large breakpoint', () => {
    const component = getComponent({ textRight: 'large' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textDisabled', () => {
    const component = getComponent({ textDisabled: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textEllipsis', () => {
    const component = getComponent({ textEllipsis: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textError', () => {
    const component = getComponent({ textError: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textStrong', () => {
    const component = getComponent({ textStrong: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textSubtle', () => {
    const component = getComponent({ textSubtle: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textSuccess', () => {
    const component = getComponent({ textSuccess: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textWarning', () => {
    const component = getComponent({ textWarning: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders textWeak', () => {
    const component = getComponent({ textWeak: true });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders theme dark', () => {
    const component = getComponent({ theme: 'dark' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders theme light', () => {
    const component = getComponent({ theme: 'light' });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
