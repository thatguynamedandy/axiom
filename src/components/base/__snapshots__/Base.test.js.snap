exports[`Base renders class Component 1`] = `<span />`;

exports[`Base renders className 1`] = `
<div
  className="Lorem Ipsum" />
`;

exports[`Base renders functional Component 1`] = `<span />`;

exports[`Base renders hiddenUntil always 1`] = `
<div
  className="ax-hidden-until--true" />
`;

exports[`Base renders hiddenUntil passed large breakpoint 1`] = `
<div
  className="ax-hidden-until--large" />
`;

exports[`Base renders hiddenUntil passed medium breakpoint 1`] = `
<div
  className="ax-hidden-until--medium" />
`;

exports[`Base renders hiddenUntil passed small breakpoint 1`] = `
<div
  className="ax-hidden-until--small" />
`;

exports[`Base renders string Component 1`] = `
<span
  className="" />
`;

exports[`Base renders textBreak all 1`] = `
<div
  className="ax-text--break-all" />
`;

exports[`Base renders textBreak none 1`] = `
<div
  className="ax-text--break-none" />
`;

exports[`Base renders textBreak word 1`] = `
<div
  className="ax-text--break-word" />
`;

exports[`Base renders textCase capital 1`] = `
<div
  className="ax-text--capitalize" />
`;

exports[`Base renders textCase lower 1`] = `
<div
  className="ax-text--lowercase" />
`;

exports[`Base renders textCase upper 1`] = `
<div
  className="ax-text--uppercase" />
`;

exports[`Base renders textCenter always 1`] = `
<div
  className="ax-text--center" />
`;

exports[`Base renders textCenter passed large breakpoint 1`] = `
<div
  className="ax-text--center--large" />
`;

exports[`Base renders textCenter passed medium breakpoint 1`] = `
<div
  className="ax-text--center--medium" />
`;

exports[`Base renders textCenter passed small breakpoint 1`] = `
<div
  className="ax-text--center--small" />
`;

exports[`Base renders textDisabled 1`] = `
<div
  className="ax-text--disabled" />
`;

exports[`Base renders textEllipsis 1`] = `
<div
  className="ax-text--ellipsis" />
`;

exports[`Base renders textError 1`] = `
<div
  className="ax-text--error" />
`;

exports[`Base renders textLeft always 1`] = `
<div
  className="ax-text--left" />
`;

exports[`Base renders textLeft passed large breakpoint 1`] = `
<div
  className="ax-text--left--large" />
`;

exports[`Base renders textLeft passed medium breakpoint 1`] = `
<div
  className="ax-text--left--medium" />
`;

exports[`Base renders textLeft passed small breakpoint 1`] = `
<div
  className="ax-text--left--small" />
`;

exports[`Base renders textRight always 1`] = `
<div
  className="ax-text--right" />
`;

exports[`Base renders textRight passed large breakpoint 1`] = `
<div
  className="ax-text--right--large" />
`;

exports[`Base renders textRight passed medium breakpoint 1`] = `
<div
  className="ax-text--right--medium" />
`;

exports[`Base renders textRight passed small breakpoint 1`] = `
<div
  className="ax-text--right--small" />
`;

exports[`Base renders textStrong 1`] = `
<div
  className="ax-text--strong" />
`;

exports[`Base renders textSubtle 1`] = `
<div
  className="ax-text--subtle" />
`;

exports[`Base renders textSuccess 1`] = `
<div
  className="ax-text--success" />
`;

exports[`Base renders textWarning 1`] = `
<div
  className="ax-text--warning" />
`;

exports[`Base renders textWeak 1`] = `
<div
  className="ax-text--weak" />
`;

exports[`Base renders theme dark 1`] = `
<div
  className="ax-theme--dark" />
`;

exports[`Base renders theme light 1`] = `
<div
  className="ax-theme--light" />
`;

exports[`Base renders visibleUntil always 1`] = `
<div
  className="ax-visible-until--true" />
`;

exports[`Base renders visibleUntil passed large breakpoint 1`] = `
<div
  className="ax-visible-until--large" />
`;

exports[`Base renders visibleUntil passed medium breakpoint 1`] = `
<div
  className="ax-visible-until--medium" />
`;

exports[`Base renders visibleUntil passed small breakpoint 1`] = `
<div
  className="ax-visible-until--small" />
`;

exports[`Base renders with defaultProps 1`] = `
<div
  className="" />
`;
